service/src/main/kotlin/fi/espoo/evaka/reports/OccupancyReport.kt

```diff
@@ -289,15 +305,20 @@ private fun Database.Read.calculateOccupancyGroupReport(
         .map { (_, reportRow) -> reportRow }
 }
 
-private data class Caretakers(
-    @Nested
-    val unit: UnitKey,
+private interface Key {
+    val groupingId: UUID
+    val unitId: UUID
+}
```

Comment:
A sealed class doesn't really offer anything over using an interface since we don't do anything conditionally depending on values, just reified type parameters. I can explain this if you want.

Reply:
I think there was some when-block where the else-branch would not be needed if this was a sealed class, because compiler would then know that all options have been handled. But yeah thats a very minor benefit.

Reply:
It doesn't seem to work with type parameters, only with actual values.